pkg/trace/api/api.go

```diff
@@ -467,37 +463,34 @@ func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
 			}
 
 			newb := stats.Bucket{
-				Start:    group.Start,
-				Duration: group.Duration,
+				Start:    int64(group.Start),
+				Duration: int64(group.Duration),
 				Counts:   make(map[string]stats.Count),
 			}
-			grain, tagset := stats.AssembleGrain(&buf, b.Env, b.Resource, b.Service, tags)
+			grain, tagset := stats.AssembleGrain(&buf, out.Env, b.Resource, b.Service, tags)
 			key := stats.GrainKey(b.Name, stats.HITS, grain)
 			newb.Counts[key] = stats.Count{
-				Key:      key,
-				Name:     b.Name,
-				Measure:  stats.HITS,
-				TagSet:   tagset,
-				TopLevel: b.TopLevel,
-				Value:    b.Hits,
+				Key:     key,
+				Name:    b.Name,
+				Measure: stats.HITS,
+				TagSet:  tagset,
```

Comment:
you still need to set TopLevel on the old payload,
you can set it at Hits value
TopLevel: stats.HITS

Reply:
Done. PTAL